internal/cli/cli.go

```diff
@@ -0,0 +1,221 @@
+package cli
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	sopsconf "go.mozilla.org/sops/v3/config"
+	"go.mozilla.org/sops/v3/decrypt"
+)
+
+var SopsNoConfigMatch = errors.New("error loading config: no matching creation rules found")
+
+type RootCommand struct{}
+
+func NewRootCommand() *cobra.Command {
+	root := &RootCommand{}
+	cmd := &cobra.Command{
+		Use:               "sops-precommit",
+		PersistentPreRunE: root.persistentPreRunE,
+		RunE:              root.runE,
+	}
+	cmd.PersistentFlags().String("log-level", "info", "Configure log level")
+	return cmd
+}
+
+type decryption interface {
+	File(filepath string, ext string) ([]byte, error)
+}
+
+type sopsrules interface {
```

Comment:
This name is a bit confusing, seems like this is just a sops client interface.

Reply:
I changed it slightly to make more sense.